### PR TITLE
Different approach

### PR DESCRIPTION
**Pull request by lasote**
Hi! 
I'been thinking about the things you mentioned about the testing and I come with a new approach where the main functionality is in the `replace_in_file`.
Also I've modified the `replace_in_file` to return True or False (replaced or not replaced).
Let me know what you think. 
PD: I've not passed the whole test suite so probably I've broken something, but we could fix it later.

### CODE REVIEW

File: conans/client/tools/win.py [hunk 1]
```diff
@@ -490,3 +490,32 @@ def run_in_windows_bash(conanfile, bashcmd, cwd=None, subsystem=None, msys_mingw
         conanfile.output.info('run_in_windows_bash: %s' % wincmd)
         # https://github.com/conan-io/conan/issues/2839 (subprocess=True)
         return conanfile._conan_runner(wincmd, output=conanfile.output, subprocess=True)
+
+
+def replace_windows_paths_in_file(file_path, path_search, replace, strict=True):
```
Comment (lasote) [inline conans/client/tools/win.py]: I'm thinking about reusing the same "replace_in_file" but with `case_sensitive=True` and `pathsep_sensitive=True` new arguments

**Comment by weatherhead99** [comment]
hmmm,
well I do have some issues with this way I think..
Chiefly this: as far as I can see, the cmake config file is just about the _only_ place where you're on Windows, but want an odd unix style path that still has a drive letter but has "/" as the separator instead of "\".  So, it seems a bit of a "sledgehammer to crack a wallnut" approach to change tools.replace_in_files to provide this specific functionality. 

Secondly: I think the new options to replace_in_files are perhaps badly named, given what I said above. The "C:/cmake/style/windows/path" style path has nothing intrinsically to do with case sensitivity or path separator sensitivity.  I can see perhaps why something like this might be useful in general (especially case sensitivity), but it might be confusing to use.

**Comment by lasote** [comment]
Actually, I think it's useful for the general purpose. 
- For case sensitive: Replace any apparition of "String" without taking into account the casing of "String", for patching cmake scripts functions/other stuff where in many occasions the case doesn't matter.
- For pathsep: The present use case is a good example. It is not the first time I need to replace a path in a file and struggle with the pathsep (and casing).

I don't understand what is wrong with the naming. @jgsogo @danimtb feedback welcome

**Comment by weatherhead99** [comment]
@lasote I 100% agree that in general case insensitive options for all find/replace style things are useful!  And if this actually works to do what I was trying to achieve I'm all for it. The thing that concerns me is edge cases that we haven't run into for the path separator sensitivity.

Also, will this get accepted into upstream conan with a change this big to replace_in_files? Should this be two separate PRs to conan? (@lasote since you're a core conan guy if you say no I'll just merge this).

**Comment by lasote** [comment]
Well, it doesn't change anything in replace_in_file if the new arguments are not used, only the return code. But I think we are Ok since previously it returned a useless None. But I want my team's feedback.

**Comment by lasote** [comment]
There is no consensus in the team about it, so I will update this PR soon.

**Comment by lasote** [comment]
Updated!

**Pull request merged**